src/Storage/Flysystem2Storage.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Kevinrob\GuzzleCache\Storage;
+
+use Kevinrob\GuzzleCache\CacheEntry;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FileNotFoundException;
+
+class Flysystem2Storage implements CacheStorageInterface
+{
+
+    /**
+     * @var Filesystem
+     */
+    protected $filesystem;
+
+    public function __construct(FilesystemAdapter $adapter)
+    {
+        $this->filesystem = new Filesystem($adapter);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function fetch($key)
+    {
+        if ($this->filesystem->fileExists($key)) {
+            // The file exist, read it!
+            $data = @unserialize(
+                $this->filesystem->read($key)
+            );
+
+            if ($data instanceof CacheEntry) {
+                return $data;
+            }
+        }
+
+        return;
```

Comment:
this needs to return `NULL` not void, per `\Kevinrob\GuzzleCache\Storage\CacheStorageInterface::fetch`